src/components/EarnReport.tsx

```diff
@@ -8,172 +12,375 @@ import Balance from './Balance'
+        [SORT_KEYS.earnedAmountInSats]: (array) => array.sort((a, b) => +a.earnedAmount - +b.earnedAmount),
+      },
+    }
+  )
 
   return (
     <>
-      {empty ? (
-        <rb.Alert variant="info">{t('earn.alert_empty_report')}</rb.Alert>
-      ) : (
-        <div>
-          <rb.Table striped bordered hover variant={settings.theme} responsive>
-            <thead>
-              <tr>
-                {headers.map((name, index) => (
-                  <th key={`header_${index}`}>{reportHeadingMap[name]?.heading || name}</th>
-                ))}
-              </tr>
-            </thead>
-            <tbody>
-              {visibleLines.map((line, trIndex) => (
-                <tr key={`tr_${trIndex}`}>
-                  {line.map((val, tdIndex) => (
-                    <td key={`td_${tdIndex}`}>
-                      {headers[tdIndex] && reportHeadingMap[headers[tdIndex]]?.format === 'balance' ? (
-                        <Balance valueString={val} convertToUnit={settings.unit} showBalance={settings.showBalance} />
-                      ) : (
-                        val
-                      )}
-                    </td>
-                  ))}
-                </tr>
-              ))}
-            </tbody>
-          </rb.Table>
-          <div className="my-1 d-flex justify-content-end">
-            <small>
-              {t('earn.text_report_length', {
-                visibleLines: visibleLines.length,
-                linesWithoutHeader: linesWithoutHeader.length,
+      <Table data={tableData} theme={tableTheme} sort={tableSort} layout={{ custom: true, horizontalScroll: true }}>
+        {(tableList) => (
+          <>
+            <Header>
+              <HeaderRow>
+                <HeaderCellSort sortKey={SORT_KEYS.timestamp}>{t('earn.report.heading_timestamp')}</HeaderCellSort>
+                <HeaderCellSort sortKey={SORT_KEYS.cjTotalAmountInSats}>
+                  {t('earn.report.heading_cj_amount')}
+                </HeaderCellSort>
+                <HeaderCellSort sortKey={SORT_KEYS.inputCount}>{t('earn.report.heading_input_count')}</HeaderCellSort>
+                <HeaderCellSort sortKey={SORT_KEYS.inputAmountInSats}>
+                  {t('earn.report.heading_input_value')}
+                </HeaderCellSort>
+                <HeaderCellSort sortKey={SORT_KEYS.feeInSats}>{t('earn.report.heading_cj_fee')}</HeaderCellSort>
+                <HeaderCellSort sortKey={SORT_KEYS.earnedAmountInSats}>
+                  {t('earn.report.heading_earned')}
+                </HeaderCellSort>
+                <HeaderCell>{t('earn.report.heading_notes')}</HeaderCell>
+              </HeaderRow>
+            </Header>
+            <Body>
+              {tableList.map((item) => {
+                const entry = toEarnReportEntry(item)
+                return (
+                  <Row key={item.id} item={item}>
+                    <Cell>{entry.timestamp.toLocaleString()}</Cell>
```

Comment:
I personally prefer ISO strings for timestamps. Might be e personal thing. Aay thoughts?